src/state/routing/useRoutingAPITrade.ts

```diff
@@ -78,55 +78,46 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
     account,
     tokenIn: currencyIn,
     tokenOut: currencyOut,
-    amount: skipFetch ? undefined : amountSpecified,
+    amount: amountSpecified,
     tradeType,
     routerPreference,
   })
 
-  const {
-    isError,
-    data: tradeResult,
-    error,
-    currentData: currentTradeResult,
-  } = useGetQuoteQuery(queryArgs ?? skipToken, {
+  const { isError, data: tradeResult, error, currentData } = useGetQuoteQueryState(queryArgs)
+  useGetQuoteQuery(skipFetch ? skipToken : queryArgs, {
     // Price-fetching is informational and costly, so it's done less frequently.
     pollingInterval: routerPreference === INTERNAL_ROUTER_PREFERENCE_PRICE ? ms(`1m`) : AVERAGE_L1_BLOCK_TIME,
     // If latest quote from cache was fetched > 2m ago, instantly repoll for another instead of waiting for next poll period
     refetchOnMountOrArgChange: 2 * 60,
   })
-  const isCurrent = currentTradeResult === tradeResult
+  const isFetching = currentData !== tradeResult || !currentData
 
   return useMemo(() => {
-    if (skipFetch && amountSpecified) {
```

Comment:
q: checking that we do in fact want to no longer return the stale trade when skipping fetch but have valid input?

Reply:
fixed